cmn/tests/bench_hreq_test.go

```diff
@@ -14,23 +14,7 @@ import (
 
 // go test -bench=. -benchmem -benchtime=5s
 
-func BenchmarkWithoutHpool(b *testing.B) {
-	b.RunParallel(func(pb *testing.PB) {
-		for pb.Next() {
-			args := &cmn.HreqArgs{
-				Method: http.MethodGet,
-				Base:   "http://localhost:80",
-				Path:   "/v1/bucket",
-			}
-			_, err := args.ReqDeprecated()
-			if err != nil {
-				panic(err)
-			}
-		}
-	})
-}
-
```

Comment:
let's keep it; move `args.ReqDeprecated` to the tools package and still keep this microbench

Reply:
Is there a particular file or folder that's ideal for moving this to?

Reply:
Just maybe notice that I changed `cmn/tests/bench_hreq_test.go` and also stopped using `ReqDeprecated` for PUT. This was part of a patch to speed-up URL parsing. 

The rest can certainly wait. I'll close your PR for now. But if you have time it'd be great if you, for instance, look at https://github.com/NVIDIA/aistore/issues/212 and maybe suggest your own solution. This one is a big topic.